src/gtk/toga_gtk/widgets/internal/buttons/refresh.py

```diff
@@ -0,0 +1,190 @@
+
+        self._refresh_btn_handler = self._refresh_btn.connect(
+            "clicked",
+            lambda w: on_refresh())
+
+        self._close_btn = Gtk.Button.new_from_icon_name(
+            "close-symbolic", Gtk.IconSize.BUTTON)
+        self._close_btn.set_can_focus(False)
+        close_btn_context = self._close_btn.get_style_context()
+        close_btn_context.add_class("osd")
+        close_btn_context.add_class("toga-detailed-list-floating-buttons")
+        refresh_btn_context.remove_class("button")
+
+        self._close_btn_handler = self._close_btn.connect(
+            "clicked", 
+            lambda w: on_close())
+
+        self._hbox = Gtk.HBox()
+        self._hbox.add(self._refresh_btn)
+        self._hbox.add(self._close_btn)
+
+        self.set_transition_type(Gtk.RevealerTransitionType.CROSSFADE)
+        self.set_valign(position)
+        self.set_halign(Gtk.Align.CENTER)
+        self.set_margin_top(margin)
+        self.set_margin_bottom(margin)
+        
+        self.set_reveal_child(False)
+        self.add(self._hbox)
+
+    def show(self):
+        self.set_reveal_child(True)
+    
+    def hide(self):
+        self._refresh_btn.hide()
+        self._close_btn.hide()
+        self.set_reveal_child(False)
+
+    def show_close(self):
+        return self._close_btn.show_now()
+
+    def hide_close(self):
+        return self._close_btn.hide()
+
+    def show_refresh(self):
+        return self._refresh_btn.show_now()
+
+    def is_visible(self):
+        return self.get_reveal_child()
+
+
+class RefreshButton(ParentPosition):
+    """
+    Shows a refresh button at the top of a list when the user is at the bottom of the list.
+    Shows a refresh button at the bottom of a list when the user is at the top of the list.
+    When there is not enough content to scroll, show the button at the bottom and have a side 
+    button to move it to the top. After moving the button to the top, show a button to move it
+    to the bottom.
+
+    Example:
+     -------------
+    | Refresh | X |
+     -------------
+    """
+    def __init__(self, adj: Gtk.Adjustment, margin=12, *args, **kwargs):
+        super().__init__(adj, *args, **kwargs)
+        self.margin = margin
+        self._parent = None
+        self._on_refresh = None
+        
+        self.button_top = RefreshButtonWidget(self._on_refresh_clicked,
+                                              self._on_close_clicked,
+                                              Gtk.Align.START,
+                                              self.margin)
+
+        self.button_bottom = RefreshButtonWidget(self._on_refresh_clicked,
+                                                 self._on_close_clicked,
+                                                 Gtk.Align.END,
+                                                 self.margin)
+
+        self.adj_handler = self.adj.connect(
+            "value-changed",
+            lambda adj: self.list_changed())
```

Comment:
Same comment about `gtk_` prefixing and defining a full handler.

Reply:
`list_changed` can be called externally. `DetailedList` will call this method after, adding, removing or changing  a row.

Reply:
If `list_changed()` is called externally, then it should continue to exist as-is. What I'm saying is explicit is better than implicit: having both `list_changed()` (the public method) and `gtk_value_changed()` (an explicit handler whose implementation calls `list_changed()` is preferable to a lambda, as it's easier to reason about the structure of the code; and should the need arise, you can debug or modify the gtk handler independent of the `value-changed` signal handler.